multi-tenancy-observability/README.md

```diff
@@ -0,0 +1,270 @@
+
+We envisionned different distinct approaches here:
+
+#### Approach 1: Customers provide their logs themselves
+
+In this approach, based on the tenant information that the customer provided to us, we would generate a user/password pair that the customer would then use in their favorite log shipper to send their logs to Loki.
+
+Pros:
+- This is quite easy for us to implement as we already have the password generation logic
+- We allow customers to use the tool they know and love.
+- We will not receive pages if the tool is not working properly
+
+Cons:
+- This is far from the idea of a platform
+- There is a high risk that configuration will be messy and hard for use to debug when we need to
+
+This approach might be useful for customers that want to send logs from outside the clusters
+
+#### Approach 2: Extract tenant information from a label/PodLog
+
+Existing logging agents support setting the tenant to send to loki using a custom log label.
+We can configure our logging agents to get the logs of specific namespaces/pods if they have a label like `giantswarm.io/tenant: my-tenant` or using a PodLog CR (if applicable).
+
+Pros:
+- Quite easy to configure for logs
+
+Cons:
+- Tenant configuration will need to be configured in multiple places (MC for the read path and WC for the write path)
+- Label approach would most likely not work for metrics OOTB (this would require customer to add new labels like metrics port, path and so on).
+
+#### Approach 3: Configure promtail based on the MC tenancy CR
+
+Altough the customer UX for the read path is still unclear, we can envision a CR like a Tenant CR that would map user's roles to namespaces/deployment under a given tenant name. The logging operator would then take care of applying all that logic
+
+Pros:
+- Everything is happening on the MC and it could map to the Project CR concept that has been talked in Product for a while.
+
+Cons:
+- MC needs to be aware of the state of all WCs (has all namespaces and so on). How can customers be sure they have all logs?
+
+#### Approach 4: We enforce 1 namespace == 1 tenant with flexibility by app and possibility of opting out
+
+It is quite easy to configure 1 tenant per namespace in our current setup and we could use a giantswarm.io/logging: false on specific namespace or containers to not get the logs.
+
+Pros:
+- Customers would only have to fill out the read path configurations (unless they want to opt-out a lot)
+- With this approach, we could still have a `giantswarm.io/tenant: my-tenant` label on some apps that should have a specific tenant attached to them.
+
+Cons:
+- Our existing customers want to start slow (team by team) and opting-out for all namespaces from the start will be a hassle. An idea to avoid this would be to be able to opt-out of the multi-tenancy features with a cluster label (`giantswarm.io/tenancy: false`).
```

Comment:
We can find that out with customers. AFAIK a lot of customers have managed namespaces that are controlled by the Platform team and where they should be able to add a default opt-out label on creation as they also add other things.

Reply:
You could probably roll out slowly (and under shared control) with Policy API. We can discuss